SpatialGDK/Source/SpatialGDKEditor/Private/SnapshotGenerator/SpatialGDKEditorSnapshotGenerator.cpp

```diff
@@ -456,7 +512,10 @@ bool ValidateAndCreateSnapshotGenerationPath(FString& SavePath)
 
 bool RunUserSnapshotGenerationOverrides(Worker_SnapshotOutputStream* OutputStream)
 {
-	Worker_EntityId NextEntityId = SpatialConstants::PLACEHOLDER_ENTITY_ID_LAST + 1;
+	const USpatialGDKEditorSettings* SpatialGDKSettings = GetDefault<USpatialGDKEditorSettings>();
+
+	const int64 StartingEntityId = SpatialGDKSettings->bGeneratePlaceholderEntitiesInSnapshot ? SpatialConstants::PLACEHOLDER_ENTITY_ID_LAST + 1 : SpatialConstants::PLACEHOLDER_ENTITY_ID_FIRST;
+	Worker_EntityId NextEntityId = StartingEntityId + (SpatialGDKSettings->LaunchConfigDesc.Workers.Num() * SpatialConstants::WORKER_AUTHORITY_ASSIGNMENT_ENTITY_COUNT);
```

Comment:
Choosing a non-square for `WORKER_AUTHORITY_ASSIGNMENT_ENTITY_COUNT` maybe resulting in logic in this function overwriting existing entity IDs when adding to the snapshot. E.g. if WORKER_AUTHORITY_ASSIGNMENT_ENTITY_COUNT = 35, your casting logic in `CreateWorkerAuthorityAssignmentEntities` may still generated 36 entities. The 36th entity ID of this would be the same ID as the first override method called here? Solutions would be to just define the side length or just pass a counter only exposing GetAndIncrement() through each of these functions?